rules/azure/initial_access_consent_grant_attack_via_azure_registered_application.toml

```diff
@@ -0,0 +1,63 @@
+[metadata]
+creation_date = "2020/09/01"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/09/01"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies when a user grants permissions to an Azure-registered application or when an administrator grants tenant-wide
+permissions to an application. An adversary may create an Azure-registered application that requests access to data such
+as contact information, email, or documents. The attacker then tricks an end user into granting that application consent
+to access their data either through a phishing attack, or by injecting illicit code into a trusted website. After the
+illicit application has been granted consent, it has account-level access to data without the need for an organizational
+account. Normal remediation steps, like resetting passwords for breached accounts or requiring Multi-Factor
+Authentication (MFA) on accounts, are not effective against this type of attack, since these are third-party
+applications and are external to the organization.
+"""
+from = "now-25m"
+index = ["filebeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Possible Consent Grant Attack via Azure-Registered Application"
+note = "The Azure Filebeat module must be enabled to use this rule."
+references = [
+    "https://docs.microsoft.com/en-us/microsoft-365/security/office-365-security/detect-and-remediate-illicit-consent-grants?view=o365-worldwide",
+]
+risk_score = 47
+rule_id = "1c6a8c7a-5cb6-4a82-ba27-d5a5b8a40a38"
+severity = "medium"
+tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Identity and Access"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Consent to application"
```

Comment:
Starting to wonder why `event.outcome` isn't populating for you, but it did for me, instead, for you, `azure.activitylogs.properties.result` is populating.

Reply:
Actually looking at a couple others, looks like `event.outcome` isn't populated with `activitylogs`, interesting.

Reply:
Confirmed that `event.outcome` is missing from this event type. Weird...

Reply:
Your comment in another PR prompted me to check if `azure.activitylogs.properties.result:success` is present. It is. I'm open to adding it in, but it doesn't follow the ECS format of our other rules.

I'm wondering if the ECS mapping needs to be updated so that the Azure field  `azure.activitylogs.properties.result` maps to ECS field, `event.outcome`.

Reply:
yeah its odd because some activity logs populate it and others do not. i do think `azure.activitylogs.properties.result` should always map to it.

Reply:
The reason I would lean towards adding `azure.activitylogs.properties.result` for now is to not get a lot of failures causing the rule to fire, not sure how common it would be. @brokensound77 thoughts?

Reply:
I think there might be something wrong with the following code in the Azure module. I'm not au fait with Beats code, but I don't think this code will match a field named `azure.activitylogs.properties.result`

https://github.com/elastic/beats/blob/829c3b7dcc6365161d83a3b10f05a9f9990f36c3/x-pack/filebeat/module/azure/activitylogs/ingest/pipeline.yml#L74-L77

I think I'll do the following:

1. Use `azure.activitylogs.properties.result:success` in the rule for now
2. Figure out if the Azure module needs to be updated and, if it does, get that fixed
3. Set a reminder to circle back and update these rules when we can use `event.outcome`

Reply:
The tests fail when I add `and azure.activitylogs.properties.result:success` to the query. Confirmed the updated query matches the expected events.

```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:"Consent to application" and azure.activitylogs.properties.result:success
```

<img width="2558" alt="image" src="https://user-images.githubusercontent.com/56409778/92257378-5c560a80-ee92-11ea-9d3e-f9dbc60943da.png">

Reply:
PR https://github.com/elastic/beats/pull/20998 has been merged to ensure that field `azure.activitylogs.properties.result` is renamed to `event.outcome`. This PR can be merged. When a Filebeat 7.10 BC is available, I'll make this it's working as expected.